http2/src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2SampleResult.java

```diff
@@ -1,432 +1,286 @@
-            // Include request body if it can have one
-            if (!METHODS_WITHOUT_BODY.contains(method)) {
-                sb.append("\n").append(method).append(" data:\n");
-                sb.append(queryString);
-                sb.append('\n');
-            }
-            if (cookies.length() > 0) {
-                sb.append("\nCookie Data:\n");
-                sb.append(cookies);
-            } else {
-                sb.append("\n[no cookies]");
-            }
-            sb.append('\n');
-        }
-        final String sampData = super.getSamplerData();
-        if (sampData != null) {
-            sb.append(sampData);
-        }
-        return sb.toString();
-    }
-
-    /**
-     * @return cookies as a string
-     */
-    public String getCookies() {
-        return cookies;
-    }
-
-    /**
-     * @param string representing the cookies
-     */
-    public void setCookies(String string) {
-        if (string == null) {
-            cookies = "";// $NON-NLS-1$
-        } else {
-            cookies = string;
-        }
-    }
-
-    /**
-     * Fetch the query string
-     *
-     * @return the query string
-     */
-    public String getQueryString() {
-        return queryString;
-    }
-
-    /**
-     * Save the query string
-     *
-     * @param string the query string
-     */
-    public void setQueryString(String string) {
-        if (string == null) {
-            queryString = "";// $NON-NLS-1$
-        } else {
-            queryString = string;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HTTP2SampleResult extends HTTPSampleResult {
+
+  private int embebedResultsDepth;
+  private boolean embebedResults;
+  private String embeddedUrlRE;
+  private HttpFields httpFieldsResponse;
+  private boolean secondaryRequest;
+  private boolean pendingResponse;
+  private transient ListenerNotifier listenerNotifier = new ListenerNotifier();
+  private transient JMeterContext threadContext;
+  private transient JMeterVariables threadVars;
+  private transient List<SampleListener> sampleListeners;
+  private transient List<Assertion> assertions;
+  private transient List<PostProcessor> postProcessors;
+
+  private transient boolean isSync;
+
+  private static final Logger LOG = LoggerFactory.getLogger(HTTP2SampleResult.class);
+
+
+  static {
+    registerHTTP2ResultConverter();
+  }
+
+  private static void registerHTTP2ResultConverter() {
+    try {
+      Method method = SaveService.class
+          .getDeclaredMethod("registerConverter", String.class, XStream.class, boolean.class);
+      method.setAccessible(true);
+      Field target = SaveService.class.getDeclaredField("JTLSAVER");
+      target.setAccessible(true);
+      XStream JTLSAVER = (XStream) target.get(null);
+      method.invoke(null, HTTP2ResultConverter.class.getCanonicalName(), JTLSAVER, true);
+    } catch (NoSuchMethodException | NoSuchFieldException | InvocationTargetException | IllegalAccessException e) {
```

Comment:
you can use one super class instead of 4 different
`} catch (ReflectiveOperationException e) {`